src/context/WidgetSettingContext.tsx

```diff
@@ -8,18 +8,19 @@ import React, {
 } from 'react';
 
 import { useConstantState } from './ConstantContext';
-import { getWidgetSetting } from '../libs/api/widgetSetting';
+import { widgetSettingHandler } from '../libs/api/widgetSetting';
 import {
   getWidgetSessionCache,
   saveWidgetSessionCache,
+  WidgetSessionCache,
 } from '../libs/storage/widgetSessionCache';
-import { getDateNDaysLater, isPastTime } from '../utils';
+import { getDateNDaysLater, isPastTime, noop } from '../utils';
 
 interface WidgetSession {
   strategy: 'auto' | 'manual';
   userId: string;
   expireAt: number;
-  // channelUrl is optional and is dynamically generated for a manual strategy.
+  // channelUrl is optional and is dynamically generated for a legacy manual strategy.
```

Comment:
why has this become legacy? and what is new manual strategy? is there reference we can read?